app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php

```diff
@@ -0,0 +1,163 @@
+
+    /**
+     * Construct
+     *
+     * @param string $name
+     * @param string $primaryFieldName
+     * @param string $requestFieldName
+     * @param CollectionFactory $collectionFactory
+     * @param Configurable $configurableType
+     * @param ProductFactory $productFactory
+     * @param LocatorInterface $locator
+     * @param StoreRepositoryInterface $storeRepository
+     * @param RequestInterface $request
+     * @param array $meta
+     * @param array $data
+     * @param \Magento\Ui\DataProvider\AddFieldToCollectionInterface[] $addFieldStrategies
+     * @param \Magento\Ui\DataProvider\AddFilterToCollectionInterface[] $addFilterStrategies
+     */
+    public function __construct(
+        $name,
+        $primaryFieldName,
+        $requestFieldName,
+        CollectionFactory $collectionFactory,
+        Configurable $configurableType,
+        ProductFactory $productFactory,
+        LocatorInterface $locator,
+        StoreRepositoryInterface $storeRepository,
+        RequestInterface $request,
+        array $meta = [],
+        array $data = [],
+        array $addFieldStrategies = [],
+        array $addFilterStrategies = []
+    ) {
+        $this->_configurableType = $configurableType;
+        $this->productData = $productFactory;
+        $this->locator = $locator;
+        $this->request = $request;
+        $this->storeRepository = $storeRepository;
+        parent::__construct(
+            $name,
+            $primaryFieldName,
+            $requestFieldName,
+            $collectionFactory,
+            $addFieldStrategies,
+            $addFilterStrategies,
+            $meta,
+            $data
+        );
+    }
+
+    /**
+     * Filtered Collection
+     *
+     * @return Collection|\Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getCollection()
+    {
+        $collection = parent::getCollection();
+        $collection->addAttributeToSelect('status');
+
+        if ($this->getStore()) {
+            $collection->setStore($this->getStore());
+        }
+
+        $collection->addFieldToFilter(
+            'type_id',
+            ['in' => [
+                \Magento\Catalog\Model\Product\Type::TYPE_SIMPLE,
+                \Magento\Catalog\Model\Product\Type::TYPE_VIRTUAL,
+                \Magento\Catalog\Model\Product\Type::TYPE_DOWNLOADABLE
+            ]]
+        );
+
+        $product = $this->_getProduct();
+        foreach ((array)$this->_configurableType->getConfigurableAttributesAsArray($product) as $attribute) {
+            $collection->addAttributeToSelect($attribute['attribute_code']);
+            $collection->addAttributeToFilter($attribute['attribute_code'], ['notnull' => 1]);
+        }
+
+        return $collection;
+    }
+
+    /**
+     * Get product
+     *
+     * @return \Magento\Catalog\Model\Product
+     */
+    protected function _getProduct()
```

Comment:
method should be marked private (see [health index details ](https://testing-service.magento-community.engineering/reports/magento/magento2/pull/22725/44f04ea5-239e-437f-858c-906c984910ff/4115/MagentoHealthIndex/report.html)check) and not prefixed with an underscore